aws/cfn/bridge/resources.py

```diff
@@ -90,21 +82,69 @@ def __init__(self, name, source_file, options):
         self._update_timeout = int(options.get(_OPTION_UPDATE_TIMEOUT, timeout))
 
         # Determine our default action
-        action = options.get(_OPTION_DEFAULT_ACTION, None)
+        self._default_action = options.get(_OPTION_DEFAULT_ACTION, None)
+
+        config_provider = options.get(_OPTION_CONFIG_PROVIDER)
+        if config_provider:
+            self.override_config(config_provider)
+
+        if not self._queue_url:
+            raise ValueError(u"[%s] in '%s' is missing 'queue_url' attribute" % (name, source_file))
+
+        # Try to parse the region from the queues/sqs prefixed urls.
+        region_match = re.match(r"https?://(?:queue|sqs)\.([^.]+?)\.amazonaws\..+", self._queue_url, re.I | re.U)
+        if region_match:
+            self._region = region_match.group(1)
+
+        if not self._region:
+            raise ValueError(u"[%s] in '%s' must define 'region' attribute" % (name, source_file))
 
         # Set our actions for each type of event
-        self._create_action = options.get(_OPTION_CREATE_ACTION, action)
+        self._create_action = options.get(_OPTION_CREATE_ACTION, self._default_action)
         if not self._create_action:
             raise ValueError(u"[%s] in '%s' must define %s" % (name, source_file, _OPTION_CREATE_ACTION))
 
-        self._delete_action = options.get(_OPTION_DELETE_ACTION, action)
+        self._delete_action = options.get(_OPTION_DELETE_ACTION, self._default_action)
         if not self._delete_action:
             raise ValueError(u"[%s] in '%s' must define %s" % (name, source_file, _OPTION_DELETE_ACTION))
 
-        self._update_action = options.get(_OPTION_UPDATE_ACTION, action)
+        self._update_action = options.get(_OPTION_UPDATE_ACTION, self._default_action)
         if not self._update_action:
             raise ValueError(u"[%s] in '%s' must define %s" % (name, source_file, _OPTION_UPDATE_ACTION))
 
+    def override_config(self, config_provider):
+        providers = config_provider.rsplit('.', 2)
```

Comment:
Some people may want to use more than 2 elements as part of the class namespace.
The Config Provider should just be a class name (no method name needed)

The implementation of the Config Provider should be expected to adhere to an interface contract which is documented here
i.e. The Config Provider should provide a method with the following signature
````
def override_config(self, config)
@param config: a dictionary of default config values which may be overridden as needed
@type config: dict
@return None 
````